nuxeo-features/nuxeo-automation/nuxeo-automation-features/src/main/java/org/nuxeo/ecm/automation/core/operations/users/SuggestUserEntries.java

```diff
@@ -245,30 +238,89 @@ public Blob run() throws IOException {
+            obj.put(SuggestConstants.ID, userId);
+            obj.put(MarshallingConstants.ENTITY_FIELD_NAME, NuxeoPrincipalJsonWriter.ENTITY_TYPE);
+            obj.put(SuggestConstants.TYPE_KEY_NAME, SuggestConstants.USER_TYPE);
+            obj.put(SuggestConstants.PREFIXED_ID_KEY_NAME, NuxeoPrincipal.PREFIX + userId);
+            SuggestConstants.computeUserLabel(obj, firstLabelField, secondLabelField, thirdLabelField,
+                    hideFirstLabel, hideSecondLabel, hideThirdLabel, displayEmailInSuggestion, userId);
+            SuggestConstants.computeUserGroupIcon(obj, hideIcon);
+            if (!StringUtils.isBlank(groupRestriction)) {
+                // We need to load all data about the user particularly its groups.
+                user = userManager.getUserModel(userId);
+                UserAdapter userAdapter = user.getAdapter(UserAdapter.class);
+                List<String> groups = userAdapter.getGroups();
+                if (groups != null && groups.contains(groupRestriction)) {
+                    result.add(obj);
+                }
+            } else {
+                result.add(obj);
+            }
+        }
+        return result;
+    }
+
     /**
      * Performs a full name user search, e.g. typing "John Do" returns the user with first name "John" and last name
      * "Doe". Typing "John" returns the "John Doe" user and possibly other users such as "John Foo". Respectively,
      * typing "Do" returns the "John Doe" user and possibly other users such as "Jack Donald".
      */
-    protected DocumentModelList searchUsers() {
+    protected DocumentModelList searchUsers(long limit) {
         if (StringUtils.isBlank(prefix)) {
             // empty search term
             return userManager.searchUsers(prefix);
         }
-
-        String trimmedPrefix = prefix.trim();
         // split search term around whitespace, e.g. "John Do" -> ["John", "Do"]
-        String[] searchTerms = trimmedPrefix.split("\\s", 2);
-        return Stream.of(searchTerms)
-                     .map(userManager::searchUsers) // search on all terms, e.g. "John", "Do"
-                     // intersection between all search results to handle full name
-                     .reduce((a, b) -> {
-                         a.retainAll(b);
-                         return a;
-                     })
-                     .filter(result -> !result.isEmpty())
-                     // search on whole term to handle a whitespace within the first or last name
-                     .orElseGet(() -> userManager.searchUsers(trimmedPrefix));
+        String[] searchTerms = prefix.trim().split("\\s", 2);
+        List<Predicate> predicates = Arrays.stream(searchTerms)
+                                           .map(this::getUserSearchPredicate)
+                                           .collect(Collectors.toList());
+        DocumentModelList users = searchUsers(new MultiExpression(Operator.AND, predicates), limit);
```

Comment:
Could you please add back this comment here:
// intersection between all search results to handle full name

Reply:
Ok done.